src/audio/component.c

```diff
@@ -326,121 +326,210 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 	return 0;
 }
```

Comment:
@bkokoszx can you please add in the commit message a simple explanation of what 'component model' is? What is the current limitation.

Reply:
@dbaluta 
I've changed the name of component model and added, updated commit message and added a few more comments. Please take a look whether it satisfies you.